### PR TITLE
RC smoothing auto factor tooltip text for 4.3

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: https://github.com/betaflight/betaflight-configurator/issues/2711

### CODE REVIEW

**Comment by ctzsnooze** [comment]
This message will need version control.

The existing message is correct for 4.2.

For 4.3 we should say:
```
         "message": "Adjusts auto RC smoothing.  30 is the default.<br>Higher values smooth RC inputs more - e.g. 60 for HD freestyle or 90-120 for cinematic flying.<br>Note: Values over 50 will cause appreciable stick delay.<br>Lower values, eg 20-25, will transfer some of the RC control steps into the motor signals, slightly increasing motor heat, but will reduce RC delay slightly.  This may be useful for racing.",
```

File: src/js/tabs/receiver.js
Comment (Asizon) [inline src/js/tabs/receiver.js:635]: No need to attr here the tootltip for version < 1.44 since it is already defined and named in the html part.

**Review by Asizon — CHANGES_REQUESTED** [review]

**Review by Asizon — DISMISSED** [review]

File: src/js/tabs/receiver.js
Comment (haslinghuis) [reply src/js/tabs/receiver.js]: Fixed

**Comment by ctzsnooze** [comment]
Tested, works correctly.
Maybe update the text itself to:
```Adjusts auto RC smoothing.<br />30 is the default.<br /><br />Higher values, e.g. 60 for HD freestyle or 90-120 for cinematic flying, will give mooth RC inputs, but may cause noticeable stick delay.<br /><br />Lower values, eg 20-25, may be useful for racing, giving slightly less RC delay at the cost of slightly warmer motors.```

**Comment by McGiverGim** [comment]
I have seen that we have a lot of "default" values in the tooltips for Betaflight 4.3. This that seems a good idea will be a nightmare of maintenance in the future, when something "changes" and we start recommending different values.
And any change in the text makes that all the translators of all the languages need to translate it again.

**Comment by haslinghuis** [comment]
Adding a defaults struct in fc.js we could make defaults independent from translated messages.

**Comment by haslinghuis** [comment]
@chmelevskij please check as it's working but I don't like / understand the console output:
```
localization-04da3b28.js:132 i18next::interpolator: missed to pass in variable autoFactor for interpolating Adjusts the amount of auto RC smoothing.<br><br>The default is {{autoFactor}}.<br><br>Higher values, e.g. 60 for HD freestyle or 90-120 for cinematic flying, will smooth out sharp RC inputs, giving smoother video, but may cause noticeable stick delay.<br><br>Lower values, eg 20-25, may be useful for racing, giving slightly less RC delay at the cost of slightly warmer motors.
```

File: src/js/tabs/receiver.js
Comment (haslinghuis) [reply src/js/tabs/receiver.js:638]: Now changed the values to be defaults using version control to address translation work

File: src/tabs/receiver.html
Comment (chmelevskij) [inline src/tabs/receiver.html:317]: @haslinghuis I think you are getting that error because of the `i18n_title="receiverRcSmoothingAutoFactorHelp"` attribute, try removing it and it should go away.

File: src/js/tabs/receiver.js
Comment (ctzsnooze) [reply src/js/tabs/receiver.js:638]: The other numbers, where we mention higher values, won't be correct for 4.2.
I think it is better to have entirely separate tooltips for 4.2 and 4.3.
These situations arise from time to time.
We shouldn't change the 4.2 tooltip.
We probably have a similar issue with antigravity gain. In 4.2 it could be pushed to 9 or 10 and would only increase I. In 4.3 it increases P also, and has inherently higher gain, and typically should NOT be pushed to 9 or 10. I haven't had time to check all these tooltips but we will need separate tooltips, or a tooltip that explains "in 4.2, do X, in 4.3 do Y".

**Comment by ctzsnooze** [comment]
@McGiverGim tooltips will need to change when the a parameter becomes functionally different.  By that I mean, it works a bit differently, or needs to be adjusted in a different way, compared to earlier versions.
This is relatively infrequent, but when it happens, the user needs a tooltip that is relevant to the code they are running, and doesn't give them bad advice.
In 4.3 this applies to rc smoothing, antigravity, and some other things.
We don't need explicitly to include default values in tooltips, but it is very helpful to include advice on how to set the parameter to a range that works effectively.  
With Presets, a user may get values that they did not enter.  They then will wonder if the value is at default or not. 
I think it would be very helpful to be able to include the default value in most tooltips, or to have some other generic means to show what the default value is.  Having some method to easily display the default could be very useful.  If the text for "Default: " could be referenced to a specific message, and the value derived cleverly, there would be no translation effort.
However in practice when we do make changes of these kinds, we will need either a singe  'combined' tooltip in the form, "for firmware after N.N, our advice is P, for earlier firmware, our advice is Q", or, we provide separate tooltips so for 4.2 "our advice is P" and for 4.3 "our advice is Q".
The second seems better to me, especially where significant changes arise.

File: src/js/tabs/receiver.js
Comment (haslinghuis) [reply src/js/tabs/receiver.js:638]: We are free to add some extra defaults for those values too.

Comment (Asizon) [reply src/js/tabs/receiver.js:638]: This need a lotof maintain @ctzsnooze , we should use generic tooltips for all versions

**Review by Asizon — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by limonspb — DISMISSED** [review]

File: src/js/tabs/receiver.js
Comment (ctzsnooze) [reply src/js/tabs/receiver.js:638]: I agree, I think it would be simpler to just have the text for the tooltip.  I don't think we need the code that pulls the default value by version.  Just change the tooltip, if needed, by version.  
If we think about antigravity, say, at one version we make a big change, then for many versions after, most likely no further change.
But when we do make a change, it is not just the default value that changes, but perhaps the way the code works, or the tuning information.
So my feeling is that we should keep tooltips as simple as possible, but version control them so that when they change significantly, we make a new tooltip for the newer version.
We can't have the same tooltip for all versions, it would just be useless in many settings.  Eg transition.  Now we have jitter reduction we don't recommend transition anymore, so we should change that tooltip to explain the new situation in 4.3, and that tooltip would be different from the tooltip for transition before 4.3

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by ctzsnooze — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Pull request merged**
